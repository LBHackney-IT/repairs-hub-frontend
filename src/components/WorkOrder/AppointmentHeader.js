import PropTypes from 'prop-types'
import { longMonthWeekday } from '@/utils/date'

const AppointmentHeader = ({ workOrder }) => {
  return (
    <div className="mobile-working-title-banner">
      {workOrder.appointment ? (
        <h3 className="lbh-heading-h3">
          {longMonthWeekday(workOrder.appointment.date, {
            commaSeparated: false,
          })}
          <br />
          {`${workOrder.appointment.start} â€“ ${workOrder.appointment.end}`}
        </h3>
      ) : (
        <h3 className="lbh-heading-h3">No appointment</h3>
      )}
    </div>
  )
}

AppointmentHeader.propTypes = {
  workOrder: PropTypes.object.isRequired,
}

export default AppointmentHeader
