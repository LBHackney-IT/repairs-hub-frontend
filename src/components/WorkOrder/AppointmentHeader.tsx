import { longMonthWeekday } from '@/utils/date'
import { Appointment } from '@/models/Appointment'

interface Props {
  appointment: Appointment
}

const AppointmentHeader = ({ appointment }: Props) => {
  return (
    <div className="mobile-working-title-banner">
      {appointment ? (
        <h2 className="lbh-heading-h2">
          {longMonthWeekday(appointment?.date, {
            commaSeparated: false,
          })}
          <br />
          {`${appointment?.start} â€“ ${appointment?.end}`}
        </h2>
      ) : (
        <h2 className="lbh-heading-h2">No appointment</h2>
      )}
    </div>
  )
}

export default AppointmentHeader
