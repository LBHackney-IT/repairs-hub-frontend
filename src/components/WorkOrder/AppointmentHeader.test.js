import { render } from '@testing-library/react'
import AppointmentHeader from './AppointmentHeader'

describe('AppointmentHeader component', () => {
  it('should render the date when the appointment has one', () => {
    const { asFragment } = render(
      <AppointmentHeader
        appointment={{
          date: '2021-03-19',
          description: 'PM Slot',
          end: '18:00',
          start: '12:00',
        }}
      />
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render info text when there is no appointment', () => {
    const { asFragment } = render(<AppointmentHeader workOrder={{}} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
