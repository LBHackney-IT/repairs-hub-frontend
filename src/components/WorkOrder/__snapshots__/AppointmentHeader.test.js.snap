// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppointmentHeader component should render info text when there is no appointment 1`] = `
<DocumentFragment>
  <div
    class="govuk-panel govuk-panel--confirmation lbh-panel appointment-header"
  >
    <h3
      class="lbh-heading-h3"
    >
      No appointment
    </h3>
  </div>
</DocumentFragment>
`;

exports[`AppointmentHeader component should render the date when the appointment has one 1`] = `
<DocumentFragment>
  <div
    class="govuk-panel govuk-panel--confirmation lbh-panel appointment-header"
  >
    <h3
      class="lbh-heading-h3"
    >
      Friday, 19 March
      <br />
      12:00-18:00
    </h3>
  </div>
</DocumentFragment>
`;
